DataFormats/CTPPSReco/interface/CTPPSPixelLocalTrack.h

```diff
@@ -0,0 +1,188 @@
+    inline void addHit(unsigned int detId, const CTPPSPixelFittedRecHit &hit)
+    {
+      track_hits_vector_.find_or_insert(detId).push_back(hit);
+      if(hit.getIsUsedForFit()) ++numberOfPointUsedForFit_;
+    }
+
+    inline float getX0() const { return track_params_vector_[0]; }
+    inline float getX0Sigma() const { return sqrt(covarianceMatrixElement(0, 0)); }
+    inline float getX0Variance() const { return covarianceMatrixElement(0, 0); }
+
+    inline float getY0() const { return track_params_vector_[1]; }
+    inline float getY0Sigma() const { return sqrt(covarianceMatrixElement(1, 1)); }
+    inline float getY0Variance() const { return covarianceMatrixElement(1, 1); }
+
+    inline float getZ0() const { return z0_; }
+    inline void setZ0(float z0) { z0_ = z0; }
+
+    inline float getTx() const { return track_params_vector_[2]; }
+    inline float getTxSigma() const { return sqrt(covarianceMatrixElement(2, 2)); }
+
+    inline float getTy() const { return track_params_vector_[3]; }
+    inline float getTySigma() const { return sqrt(covarianceMatrixElement(3, 3)); }
+
+    inline math::GlobalVector getDirectionVector() const
+    {
+      math::GlobalVector vect(getTx(), getTy(), 1);
+      return vect.unit();
+    }
+
+    ParameterVector getParameterVector() const;
+
+    CovarianceMatrix getCovarianceMatrix() const;
+
+    inline float getChiSquared() const { return chiSquared_; }
+
+    inline float getChiSquaredOverNDF() const { 
+      if(numberOfPointUsedForFit_<= 2.) return -999.;
+      else return chiSquared_ / (2*numberOfPointUsedForFit_ - dimension); 
+    }
+
+    inline int getNDF() const {return (2*numberOfPointUsedForFit_ - dimension); }
+
+    /// returns the point from which the track is passing by at the selected z
+    inline math::GlobalPoint getTrackPoint(float z) const 
+    {
+      float delta_z = z - z0_;
+      return math::GlobalPoint(
+        track_params_vector_[0] + track_params_vector_[2] * delta_z,
+        track_params_vector_[1] + track_params_vector_[3] * delta_z,
+        z);
+    }
+
+    inline math::GlobalPoint getTrackCentrePoint()
+    {
+      return math::GlobalPoint(track_params_vector_[0], track_params_vector_[1], z0_);
+    }
+
+    math::Matrix<2,2>::type trackPointInterpolationCovariance(float z) const;
+
+    inline bool isValid() const { return valid_; }
+
+    inline void setValid(bool valid) { valid_ = valid; }
+
+    bool operator< (const CTPPSPixelLocalTrack &r);
+    
+  private:
+    inline const float& covarianceMatrixElement(int i, int j) const
+    {
+      return par_covariance_matrix_[i * dimension + j];
+    }
+
+    inline float& covarianceMatrixElement(int i, int j)
+    {
+      return par_covariance_matrix_[i * dimension + j];
+    }
+
+    edm::DetSetVector<CTPPSPixelFittedRecHit> track_hits_vector_;
+
+    /// track parameters: (x0, y0, tx, ty); x = x0 + tx*(z-z0) ...
+    float track_params_vector_[dimension];
```

Comment:
both the parameter vector and covariance can be saved as ParameterVector CovarianceMatrix types.
Is there a good reason to reduce to the plain array and pay the cost of conversion and access by copy?